tests/nfs_invalid_backend_sc_topology.go

```diff
@@ -0,0 +1,324 @@
+		accessModes          = []corev1.PersistentVolumeAccessMode{corev1.ReadWriteMany}
+		nodeAffinityKeys     = []string{"kubernetes.io/hostname"}
+		capacity             = "2Gi"
+		scName               = "nfs-sc-different-node-affinity"
+
+		// NFS Provisioner related options
+		openebsNamespace            = "openebs"
+		nfsProvisionerLabel         = "openebs.io/component-name=openebs-nfs-provisioner"
+		nfsProvisionerContainerName = "openebs-provisioner-nfs"
+		nodeAffinityKeyValues       map[string][]string
+		backendScName               = "different-node-affinity"
+		backendScNodeAffinityLabel  = "invalid.io/invalid-nfs-label"
+		backendPVCName              = ""
+	)
+
+	When("node affinity environment variable is added", func() {
+		It("should be applied", func() {
+			nodeList, err := Client.listNodes("")
+			Expect(err).To(BeNil(), "failed to list nodes")
+			var nodeAffinityAsValue string
+
+			nodeAffinityKeyValues = make(map[string][]string, len(nodeAffinityKeys))
+			// Form affinity rules from multiple nodes
+			for _, node := range nodeList.Items {
+				for _, key := range nodeAffinityKeys {
+					if value, isExist := node.Labels[key]; isExist {
+						nodeAffinityKeyValues[key] = append(nodeAffinityKeyValues[key], value)
+					}
+				}
+			}
+
+			for key, values := range nodeAffinityKeyValues {
+				nodeAffinityAsValue += key + ":["
+				for _, value := range values {
+					nodeAffinityAsValue += value + ","
+				}
+				// remove extra comma
+				nodeAffinityAsValue = nodeAffinityAsValue[:len(nodeAffinityAsValue)-1]
+				nodeAffinityAsValue += "],"
+			}
+			// remove extra comma and add key as affinity rules
+			nodeAffinityAsValue = nodeAffinityAsValue[:len(nodeAffinityAsValue)-1] + ",kubernetes.io/arch"
+			nodeAffinityKeyValues["kubernetes.io/arch"] = []string{}
+
+			deploymentList, err := Client.listDeployments(openebsNamespace, nfsProvisionerLabel)
+			Expect(err).To(BeNil(), "failed to list NFS Provisioner deployments")
+
+			nfsProvisionerDeployment := deploymentList.Items[0]
+			for index, containerDetails := range nfsProvisionerDeployment.Spec.Template.Spec.Containers {
+				if containerDetails.Name == nfsProvisionerContainerName {
+					nfsProvisionerDeployment.Spec.
+						Template.
+						Spec.
+						Containers[index].Env = append(
+						nfsProvisionerDeployment.Spec.
+							Template.
+							Spec.
+							Containers[index].Env,
+						corev1.EnvVar{
+							Name:  string(provisioner.NodeAffinityKey),
+							Value: nodeAffinityAsValue,
+						},
+					)
+					break
+				}
+			}
+
+			err = Client.applyDeployment(&nfsProvisionerDeployment)
+			Expect(err).To(BeNil(), "failed to add %s env to NFS Provisioner", provisioner.NodeAffinityKey)
+		})
+	})
+
+	When(fmt.Sprintf("create backend storageclass %s with nodeAffinityLabel=%s", backendScName, backendScNodeAffinityLabel), func() {
+		It("should create backend storageclass", func() {
+			By("creating storageclass")
+			casObj := []mayav1alpha1.Config{
```

Comment:
Where are we specifying topology on backend SC? than the provisioned affinity rules?